src/gui/guiaddtorrentmanager.cpp

```diff
@@ -181,7 +181,16 @@ bool GUIAddTorrentManager::processTorrent(const QString &source, const BitTorren
     if (!hasMetadata)
         btSession()->downloadMetadata(torrentDescr);
 
+#ifndef Q_OS_MACOS
     auto *dlg = new AddNewTorrentDialog(torrentDescr, params, app()->mainWindow());
+#else
+    //By not setting a parent to the "addnewtorrentdialog",
+    //all those dialogs will be displayed on top and will not overlap with the main window.
+    auto *dlg = new AddNewTorrentDialog(torrentDescr, params, nullptr);
+    //required Qt::Window to avoid only show two dialog on top, see #12852.
+    dlg->setWindowFlags(Qt::Window);
```

Comment:
This turns the dialog into a regular window, i.e. it can now have other properties or behavior, e.g. have Minimize/Maximize buttons, etc. Can you confirm that this does not cause undesirable consequences?

Reply:
When the dialog is enlarged(previously achieved by double-clicking the title bar) , the UI displays correctly, and the widgets stretch to the left and right. The layout is reasonable.
The title bar now includes a minimize button, but minimizing the "AddNewTorrentDialog" does not affect the functionality of other features.

Reply:
Ok.

Reply:
I wonder how it would behave if you set parent to MainWindow and this flag?

Reply:
```
    auto *dlg = new AddNewTorrentDialog(torrentDescr, params, nullptr);
    dlg->setWindowFlags(Qt::Window);
```
work fine.

```
    auto *dlg = new AddNewTorrentDialog(torrentDescr, params, app()->mainWindow());
    dlg->setWindowFlags(Qt::Window);
```
All "AddNewTorrentDialog" dialogs are shown at the bottom. After opened QFileDialog, mainWindow will be forcefully shown on top.

```
    auto *dlg = new AddNewTorrentDialog(torrentDescr, params, nullptr);
    dlg->setWindowFlags(Qt::Dialog);
```
Two “AddNew” dialogs are displayed top, and the remains are shown bottom(#12852). After opened QFileDialog, mainWindow will be forcefully shown on top.(#19072)